services/ui-src/src/components/pages/Dashboard/DashboardTable.tsx

```diff
@@ -0,0 +1,214 @@
+    {reportsByState.map((report: ReportMetadataShape) => (
+      <Tr key={report.id}>
+        {/* Edit Button */}
+        {isStateLevelUser ? (
+          <EditReportButton
+            report={report}
+            openAddEditReportModal={openAddEditReportModal}
+            sxOverride={sxOverride}
+          />
+        ) : (
+          <Td></Td>
+        )}
+        {/* Report Name */}
+        <Td sx={sxOverride.programNameText}>{report.programName}</Td>
+        {/* Date Fields */}
+        <DateFields report={report} reportType={reportType} />
+        {/* Last Altered By */}
+        <Td>{report?.lastAlteredBy || "-"}</Td>
+        {/* Report Status */}
+        <Td>{report?.archived ? "Archived" : report?.status}</Td>
+        {/* MLR-ONLY: Submission count */}
+        {reportType === "MLR" && <Td></Td>}
+        {/* Action Buttons */}
+        <Td sx={sxOverride.editReportButtonCell}>
+          <Button
+            variant="outline"
+            data-testid="enter-report"
+            onClick={() => enterSelectedReport(report)}
+            isDisabled={report?.archived}
+          >
+            Enter
+          </Button>
+        </Td>
+        {isAdmin && (
+          <AdminActionButtons
+            report={report}
+            reportType={reportType}
+            archiveReport={archiveReport}
+            archiving={archiving}
+            unlockReport={unlockReport}
+            unlocking={unlocking}
+            reportId={reportId}
+            sxOverride={sxOverride}
+          />
+        )}
+      </Tr>
+    ))}
+  </Table>
+);
+
+interface DashboardTableProps {
+  reportsByState: ReportMetadataShape[];
+  body: { table: AnyObject };
+  reportType: string;
+  reportId: string | undefined;
+  openAddEditReportModal: Function;
+  enterSelectedReport: Function;
+  archiveReport: Function;
+  archiving: boolean;
+  isAdmin: boolean;
+  isStateLevelUser: boolean;
+  unlockReport?: Function | undefined;
+  unlocking?: boolean | undefined;
+  sxOverride: AnyObject;
+}
+
+const EditReportButton = ({
+  report,
+  openAddEditReportModal,
+  sxOverride,
+}: EditReportProps) => {
+  return (
+    <Td sx={sxOverride.editReport}>
+      <button onClick={() => openAddEditReportModal(report)}>
+        <Image src={editIcon} alt="Edit Report" />
+      </button>
+    </Td>
+  );
+};
+
+interface EditReportProps {
+  report: ReportMetadataShape;
+  openAddEditReportModal: Function;
+  sxOverride: AnyObject;
+}
+
+const DateFields = ({ report, reportType }: DateFieldProps) => {
+  return (
+    <>
+      {reportType === "MCPAR" && <Td>{convertDateUtcToEt(report.dueDate)}</Td>}
+      <Td>{convertDateUtcToEt(report.lastAltered)}</Td>
+    </>
+  );
+};
+
+interface DateFieldProps {
+  report: ReportMetadataShape;
+  reportType: string;
+}
+
+const AdminActionButtons = ({
+  report,
+  reportType,
+  reportId,
+  archiveReport,
+  archiving,
+  unlocking,
+  unlockReport,
+  sxOverride,
+}: AdminActionButtonProps) => {
```

Comment:
Thought: Rather than passing through archiving and unlocking functions, you could pass through the state (unlocking or archiving) and function (archiveReport or unlockReport). That way it would be report agnostic and you could just pass through the function above as needed depending on what report it is. Might make it easier in the future when we add more reports.